apps/rush-lib/src/cli/utilities/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -0,0 +1,132 @@
+import * as fsx from 'fs-extra';
+import * as yaml from 'js-yaml';
+import * as os from 'os';
+
+import Utilities from '../../../utilities/Utilities';
+import { BaseShrinkwrapFile } from '../base/BaseShrinkwrapFile';
+
+interface IShrinkwrapDependencyJson {
+  /** Information about the resolved package */
+  resolution: {
+    /** The hash of the tarball, to ensure archive integrity */
+    integrity: string;
+    /** The name of the tarball, if this was from a TGX file */
+    tarball?: string;
+  };
+  /** The list of dependencies and the resolved version */
+  dependencies: { [dependency: string]: string };
+}
+
+/**
+ * This interface represents the raw shrinkwrap.YAML file
+ * Example:
+ *  {
+ *    "dependencies": {
+ *      "@rush-temp/project1": "file:./projects/project1.tgz"
+ *    },
+ *    "packages": {
+ *      "file:projects/library1.tgz": {
+ *        "dependencies: {
+ *          "markdown": "0.5.0"
+ *        },
+ *        "name": "@rush-temp/library1",
+ *        "resolution": {
+ *          "tarball": "file:projects/library1.tgz"
+ *        },
+ *        "version": "0.0.0"
+ *      },
+ *      "markdown/0.5.0": {
+ *        "resolution": {
+ *          "integrity": "sha1-KCBbVlqK51kt4gdGPWY33BgnIrI="
+ *        }
+ *      }
+ *    },
+ *    "registry": "http://localhost:4873/",
+ *    "shrinkwrapVersion": 3,
+ *    "specifiers": {
+ *      "@rush-temp/project1": "file:./projects/project1.tgz"
+ *    }
+ *  }
+ */
+interface IShrinkwrapYaml {
+  /** The list of resolved version numbers for direct dependencies */
+  dependencies: { [dependency: string]: string };
+  /** The description of the solved graph */
+  packages: { [dependencyVersion: string]: IShrinkwrapDependencyJson };
+  /** URL of the registry which was used */
+  registry: string;
+  /** The PNPM shrinkwrap internal versioning number */
+  shrinkwrapVersion: number | undefined;
```

Comment:
>shrinkwrapVersion [](start = 2, length = 17)

If you don't use this or care about it, then remove it from the interface.

Or, if you're not 100% sure that it's defined, then simply validate that after loading it.  Declaring it as undefined means everyone who accesses it now has to validate it.